app/models/team.rb

```diff
@@ -114,43 +114,56 @@ def self.randomize_all_by_parent(parent, team_type, min_team_size)
       end
     end
     # sort teams by decreasing team size
-    teams.sort_by {|team| Team.size(team.id) }.reverse!
+    teams = sort_teams_by_members_reverse(teams)
     # insert users who are not in any team to teams still need team members
     if !users.empty? and !teams.empty?
-      teams.each do |team|
-        curr_team_size = Team.size(team.id)
-        member_num_difference = min_team_size - curr_team_size
-        for i in (1..member_num_difference).to_a
-          team.add_member(users.first, parent.id)
-          users.delete(users.first)
-          break if users.empty?
-        end
-        break if users.empty?
-      end
+      assign_single_users_to_teams(min_team_size, parent, teams, users)
     end
     # If all the existing teams are fill to the min_team_size and we still have more users, create teams for them.
     unless users.empty?
-      num_of_teams = users.length.fdiv(min_team_size).ceil
-      nextTeamMemberIndex = 0
-      for i in (1..num_of_teams).to_a
-        team = Object.const_get(team_type + 'Team').create(name: "Team" + i.to_s, parent_id: parent.id)
-        TeamNode.create(parent_id: parent.id, node_object_id: team.id)
-        min_team_size.times do
-          break if nextTeamMemberIndex >= users.length
-          user = users[nextTeamMemberIndex]
-          team.add_member(user, parent.id)
-          nextTeamMemberIndex += 1
-        end
+      create_team_from_single_users(min_team_size, parent, team_type, users)
+    end
+  end
+
+  def self.create_team_from_single_users(min_team_size, parent, team_type, users)
+    num_of_teams = users.length.fdiv(min_team_size).ceil
+    next_team_member_index = 0
+    for i in (1..num_of_teams).to_a
+      team = Object.const_get(team_type + 'Team').create(name: "Team" + i.to_s, parent_id: parent.id)
+      TeamNode.create(parent_id: parent.id, node_object_id: team.id)
+      min_team_size.times do
+        break if next_team_member_index >= users.length
+        user = users[next_team_member_index]
+        team.add_member(user, parent.id)
+        next_team_member_index += 1
       end
     end
   end
 
+  def self.assign_single_users_to_teams(min_team_size, parent, teams, users)
+    teams.each do |team|
+      curr_team_size = Team.size(team.id)
+      member_num_difference = min_team_size - curr_team_size
+      while member_num_difference > 0
+        team.add_member(users.first, parent.id)
+        users.delete(users.first)
+        member_num_difference -= 1
+        break if users.empty?
+      end
+      break if users.empty?
+    end
+  end
+
+  def self.sort_teams_by_members_reverse(teams)
```

Comment:
hmm, I don't think you need to extract one line of code as a new method. But it is not a big deal.